sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -355,20 +358,44 @@ object GpuCast extends Arm {
         castStructToString(input, fields, ansiMode, legacyCastToString,
           stringToDateAnsiModeEnabled)
 
-      // ansi cast from larger-than-integer integral types, to integer
-      case (LongType, IntegerType) if ansiMode =>
+      // ansi cast from larger-than-long integral-like types, to long
+      case (dt: DecimalType, LongType) if ansiMode =>
+        // This is a work around for https://github.com/rapidsai/cudf/issues/9282
+        val min = BigDecimal(Long.MinValue)
+            .setScale(dt.scale, BigDecimal.RoundingMode.DOWN).bigDecimal
+        val max = BigDecimal(Long.MaxValue)
+            .setScale(dt.scale, BigDecimal.RoundingMode.DOWN).bigDecimal
+        assertValuesInRange(input, Scalar.fromDecimal(min), Scalar.fromDecimal(max))
+        if (dt.precision <= DType.DECIMAL32_MAX_PRECISION && dt.scale < 0) {
+          // This is a work around for https://github.com/rapidsai/cudf/issues/9281
+          withResource(input.castTo(DType.create(DType.DTypeEnum.DECIMAL64, -dt.scale))) { tmp =>
```

Comment:
Mostly curiosity, the issue linked doesn't mention negative scales, so why do we need to negate the scale here again?

Reply:
cudf scale = - spark scale

Reply:
thanks!